diff --git a/crypto/init.c b/crypto/init.c
index 9fc0e8e..ad7bf10 100644
--- a/crypto/init.c
+++ b/crypto/init.c
@@ -126,6 +126,10 @@ static int win32atexit(void)
 }
 #endif

+/* Work around wrong _onexit() definition of some mingw-w64 versions.
+ * ld fails to find _onexit reference when --disable-auto-import is set. */
+__declspec(dllimport) _onexit_t _onexit(_onexit_t);
+
 DEFINE_RUN_ONCE_STATIC(ossl_init_register_atexit)
 {
 #ifdef OPENSSL_INIT_DEBUG
